pkg/reporterreporter/report.go

```diff
@@ -6,21 +6,29 @@ import (
 	"time"
 
 	"bitbucket.org/atlassianlabs/restclient"
+	"github.com/atlassian/ctrl"
 	reporter_v1 "github.com/atlassian/voyager/pkg/apis/reporter/v1"
 	"github.com/atlassian/voyager/pkg/reporter/client"
+	"github.com/atlassian/voyager/pkg/util"
 	"github.com/pkg/errors"
 	"go.uber.org/zap"
 	meta_v1 "k8s.io/apimachinery/pkg/apis/meta/v1"
 	"k8s.io/client-go/kubernetes"
 )
 
+const (
+	retryAttemptsPerReport = 5
+	retryDelay             = time.Second * 1
+)
+
 type Report struct {
-	RemoteURI        string
 	Cluster          string
```

Comment:
You can just make all the fields private because now it's impossible to construct a valid instance anyway - `mutator` is private.

Reply:
Shouldn't we be making Report an interface instead? Making them all private doesn't change how the ability to create one?

I could also make mutator public?